_posts/2019/2019-02-24-secure-your-staging-environment-with-basic-auth.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Secure Your Rails Staging Environment with HTTP Basic Authentication
+description: Opening up your development environment is a bad idea
+layout: article
+category: ruby
+image:
+  base: '2019/secure-your-staging-environment-with-basic-auth'
+  alt: 'Yellow padlock'
+  source: "https://unsplash.com/photos/0Yiy0XajJHQ"
+  credit: "Chris Panas"
+---
+
+A sensible approach to testing features before you ship them to customers is to use a staging environment that closely resembles your live, customer-facing, production application.
+
+It’s important to make sure that only people inside your organisation can use this testing environment, since it might contain broken features (hopefully not!) or private, production-like, data.
+
+
+## Instead of…
+
+…leaving your staging environment open for anyone who finds it on the wild Internet.
+
+
+## Use…
+
+…the basic HTTP authentication in Rails to secure the whole application from non-authorised users.
```

Comment:
Change the full stop at the end here to an ellipsis since you have the '...and' after the code example.